src/Controls/PlanarControls.js

```diff
@@ -234,183 +274,177 @@ function PlanarControls(view, options = {}) {
-                this.camera.position.z = newAltitude;
-            }
-        };
-    })();
+    handlePanMovement() {
+        const vect = new THREE.Vector3();
+
+        // normalized (between 0 and 1) distance between groundLevel and maxAltitude
+        const distToGround = THREE.MathUtils.clamp(
+            (this.camera.position.z - this.groundLevel) / this.maxAltitude,
+            0,
+            1,
+        );
+
+        // pan movement speed, adjusted according to altitude
+        const panSpeed = THREE.MathUtils.lerp(
+            this.minPanSpeed,
+            this.maxPanSpeed,
+            distToGround,
+        );
+
+        // lateral movement (local x axis)
+        vect.set(panSpeed * -1 * deltaMousePosition.x, 0, 0);
+        this.camera.position.copy(this.camera.localToWorld(vect));
+
+        // vertical movement (world z axis)
+        const newAltitude = this.camera.position.z + panSpeed * deltaMousePosition.y;
+
+        // check if altitude is valid
+        if (this.groundLevel < newAltitude && newAltitude < this.maxAltitude) {
+            this.camera.position.z = newAltitude;
+        }
+    }
 
     /**
      * Initiate a rotate (orbit) movement.
      *
      * @ignore
      */
-    this.initiateRotation = function initiateRotation() {
+    initiateRotation() {
         this.state = STATE.ROTATE;
 
+        centerPoint.copy(this.getWorldPointAtScreenXY(new THREE.Vector2(
+            0.5 * this.view.mainLoop.gfxEngine.width,
+            0.5 * this.view.mainLoop.gfxEngine.height,
+        )));
 
-        centerPoint.copy(this.getWorldPointAtScreenXY({ x: 0.5 * this.view.mainLoop.gfxEngine.width, y: 0.5 * this.view.mainLoop.gfxEngine.height }));
-
-        const r = this.camera.position.distanceTo(centerPoint);
-        phi = Math.acos((this.camera.position.z - centerPoint.z) / r);
-    };
+        const radius = this.camera.position.distanceTo(centerPoint);
+        phi = Math.acos((this.camera.position.z - centerPoint.z) / radius);
+    }
 
     /**
-     * Handle the rotate movement (orbit) when user moves the mouse while in
-     * STATE.ROTATE. The movement is an orbit around 'centerPoint', the camera
-     * focus point (ground point at screen center). The rotate movement is
-     * previously initiated in [initiateRotation]{@link PlanarControls#initiateRotation}.
+     * Handle the rotate movement (orbit) when user moves the mouse while in STATE.ROTATE. The movement is an
+     * orbit around `centerPoint`, the camera focus point (ground point at screen center). The rotate movement
+     * is previously initiated in [initiateRotation]{@link PlanarControls#initiateRotation}.
      * Compute the new position value and update the camera controls.
      *
-     * @function
      * @ignore
      */
-    this.handleRotation = (() => {
-        const vec = new THREE.Vector3();
+    handleRotation() {
+        const vect = new THREE.Vector3();
```

Comment:
could instance `vect` and `quad` before Class? (or use free variable already instanced)